src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -300,22 +368,55 @@ internal sealed partial class ForToLoopOperation
 
     internal sealed partial class WhileLoopOperation
     {
-        public override IEnumerable<IOperation> Children
+        protected override IOperation GetCurrent(int slot, int index)
+            => (slot, ConditionIsTop) switch
```

Comment:
> => (slot, ConditionIsTop) switch [](start = 11, length = 33)

I think the logic will be much easier to follow if we separated two cases based on ConditionIsTop rather than trying to "jam" them togather.